api/src/main/java/org/openmrs/validator/ConditionValidator.java

```diff
@@ -46,9 +48,11 @@ public void validate(Object object, Errors errors) {
 		Condition condition = (Condition) object;
 		if(condition.getCondition() == null){
 			errors.rejectValue("condition", "Condition.conditionShouldNotBeNull", "The condition is required");
+		} else if(condition.getClinicalStatus().equals(ConditionClinicalStatus.INACTIVE) && condition.getOnsetDate().compareTo(condition.getEndDate()) > 0){
+			errors.rejectValue("onsetDate","Condition.onsetDateShouldBeBeforeEndDate","Valid Onset Date is required");
```

Comment:
Can you add a test for this?

Reply:
I added shouldPassForValidOnsetDate() to test this in ConditionValidatorTest.java

Reply:
How is it testing the above line?

Reply:
I added another test function shouldFailForInvalidOnsetDate() and updated the PR :)

Reply:
@haripriya999 where's this `shouldFailForInvalidOnsetDate()`? I only see `shouldPassWhenOnsetDateIsBeforeEndDate()` that seem to be testing both the 'pass' and 'fail' cases.

Reply:
@mks-d  I wrote the code for both the test's in the same function (shouldPassWhenOnsetDateIsBeforeEndDate) in different blocks, as writing two separate test cases is causing a Coveralls error. Merging the two tests into one function fixed the Coveralls error.

Reply:
Then call it `shouldValidateThatOnsetDateIsBeforeEndDate()`.

Can you still point us to the Coveralls error trace?

Also don't force push, just add commits to your PR.

Reply:
Thank you @mks-d .Pushing it as separate commits isn't causing the coveralls issue even after putting the code in separate function i.e shouldFailForInvalidOnsetDate() .  Please review the changes. :)